File: money_manager_app/lib/features/wallet/domain/entities/wallet.dart
import 'package:equatable/equatable.dart';

class Wallet extends Equatable {
  final String id;
  final String name;
  final double initialBalance; // Số dư ban đầu khi tạo ví
  final double balance; // Số dư hiện tại = initialBalance + income - expense
  final String currency;
  final String type;
  final bool isDeleted;
  final DateTime lastUpdatedAt;
  final bool isSynced;

  const Wallet({
    required this.id,
    required this.name,
    this.initialBalance = 0,
    this.balance = 0,
    this.currency = 'VND',
    this.type = 'CASH',
    this.isDeleted = false,
    required this.lastUpdatedAt,
    this.isSynced = false,
  });

  Wallet copyWith({
    String? id,
    String? name,
    double? initialBalance,
    double? balance,
    String? currency,
    String? type,
    bool? isDeleted,
    DateTime? lastUpdatedAt,
    bool? isSynced,
  }) {
    return Wallet(
      id: id ?? this.id,
      name: name ?? this.name,
      initialBalance: initialBalance ?? this.initialBalance,
      balance: balance ?? this.balance,
      currency: currency ?? this.currency,
      type: type ?? this.type,
      isDeleted: isDeleted ?? this.isDeleted,
      lastUpdatedAt: lastUpdatedAt ?? this.lastUpdatedAt,
      isSynced: isSynced ?? this.isSynced,
    );
  }

  @override
  List<Object?> get props => [id, name, initialBalance, balance, currency, type, isDeleted, lastUpdatedAt, isSynced];
}

File: MoneyManager-master/MoneyManager.Domain/Entities/Wallet.cs
using MoneyManager.Domain.Enums;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace MoneyManager.Domain.Entities;

public partial class Wallet : BaseSyncEntity
{
    public string Name { get; set; } = string.Empty;

    /// <summary>
    /// Số dư ban đầu khi tạo ví
    /// </summary>
    [Column(TypeName = "decimal(18, 2)")]
    public decimal InitialBalance { get; set; } = 0;

    /// <summary>
    /// Số dư hiện tại = InitialBalance + Tổng Thu (Income) - Tổng Chi (Expense)
    /// </summary>
    [Column(TypeName = "decimal(18, 2)")]
    public decimal? Balance { get; set; }
    
    public CurrencyCode Currency { get; set; } = CurrencyCode.VND;

    public string Type { get; set; } = null!;

    public Guid OwnerId { get; set; }
    [ForeignKey("OwnerId")]
    public virtual AppUser Owner { get; set; } = null!;

    public virtual ICollection<Transaction> Transactions { get; set; } = new List<Transaction>();
}
